### ISSUE TITLE
Release 4.3.0 - Release candidate 6 - Manual tests - Active response

### ISSUE THREAD

**Issue by vikman90** [wazuh/wazuh#13222]
The following issue aims to run the specified test for the current release candidate, report the results, and open new issues for any encountered errors.

## Test information

|Test name|Test|Deployment option|Main RC issue|Version|
|---|---|---|---|---|
|Active response|Active response|OVA|#13154|4.3.0 RC6|

## Test description

Test that the following proof of concept cases work in 4.3.0:
- https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html
- https://documentation.wazuh.com/current/proof-of-concept-guide/detect-remove-malware-virustotal.html
- https://documentation.wazuh.com/current/proof-of-concept-guide/detect-malware-yara-integration.html

Test that this blog post still works well with 4.3.0:
- https://wazuh.com/blog/how-to-integrate-yara-with-wazuh

## Test report procedure

All test results must have one of the following statuses: 
|                                  |                                            |
|---------------------------------|--------------------------------------------|
| :green_circle:  | All checks passed. |
| :red_circle:  | There is at least one failed check. |
| :yellow_circle:  | There is at least one expected failure or skipped test and no failures. |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. 

An extended report of the test results must be attached as a ZIP or TXT file. Please attach any documents, screenshots or tables to the issue update with the results. This report can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

After the review of the proof of concept and use cases done in this issue's comments, the following issues were opened and should be reviewed and taken into account for this Wazuh release.

|Status|Test|Failure type|Notes|
|---|---|---|---|
|🟢|[Blocking a malicious actor](https://github.com/wazuh/wazuh/issues/13222#issuecomment-1110905539)|-|-|
|🟢|[Detecting and removing malware using VirusTotal integration](https://github.com/wazuh/wazuh/issues/13222#issuecomment-1110908937)|-|-|
|🟢|[Detecting malware using Yara integration](https://github.com/wazuh/wazuh/issues/13222#issuecomment-1110929401)|-|https://github.com/wazuh/wazuh-documentation/pull/5134|
|⚪|[How to integrate YARA with Wazuh](https://github.com/wazuh/wazuh/issues/13222#issuecomment-1112159022)|-|-|

## Auditors' validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @TomasTurina

**Comment by vikman90** [comment]
## 🟢 Blocking a malicious actor

<details><summary>Agent setup</summary>

### Agent installation

```shell
wget https://packages-dev.wazuh.com/pre-release/yum/wazuh-agent-4.3.0-1.x86_64.rpm
yum install ./wazuh-agent-4.3.0-1.x86_64.rpm
yum install httpd
service httpd start
```

### Agent configuration

```xml
<localfile>
  <log_format>apache</log_format>
  <location>/var/log/httpd/access_log</location>
</localfile>
```

</details>
<details><summary>Manager setup</summary>

### Manager installation

```shell
wget https://raw.githubusercontent.com/firehol/blocklist-ipsets/master/alienvault_reputation.ipset -O /var/ossec/etc/lists/alienvault_reputation.ipset
echo "192.168.33.1" >> /var/ossec/etc/lists/alienvault_reputation.ipset
wget https://wazuh.com/resources/iplist-to-cdblist.py -O /tmp/iplist-to-cdblist.py
python /tmp/iplist-to-cdblist.py /var/ossec/etc/lists/alienvault_reputation.ipset /var/ossec/etc/lists/blacklist-alienvault
rm -rf /var/ossec/etc/lists/alienvault_reputation.ipset
rm -rf /var/ossec/etc/lists/iplist-to-cdblist.py
chown wazuh:wazuh /var/ossec/etc/lists/blacklist-alienvault
chmod 660 /var/ossec/etc/lists/blacklist-alienvault
systemctl restart wazuh-manager
```

### Rule

```xml
<group name="attack,">
  <rule id="100100" level="10">
    <if_group>web|attack|attacks</if_group>
    <list field="srcip" lookup="address_match_key">etc/lists/blacklist-alienvault</list>
    <description>IP address found in AlienVault reputation database.</description>
  </rule>
</group>
```

### Manager configuration

```xml
<ruleset>
  <list>etc/lists/blacklist-alienvault</list>
</ruleset>

<active-response>
  <command>firewall-drop</command>
  <location>local</location>
  <rules_id>100100</rules_id>
  <timeout>60</timeout>
</active-response>
```

</details>

### Results

#### Discover query screenshot

![discover-iprep](https://user-images.githubusercontent.com/10536251/165510748-0d5c1357-5899-43dd-940a-0534c83c5480.jpg)

<details><summary>Access alert</summary>

```json
{
  "_index": "wazuh-alerts-4.x-2022.04.27",
  "_type": "_doc",
  "_id": "GVWWaoABKmeKtYx2GU13",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "192.168.182.133",
      "name": "centos7",
      "id": "001"
    },
    "manager": {
      "name": "wazuh-manager"
    },
    "data": {
      "protocol": "GET",
      "srcip": "192.168.33.1",
      "id": "404",
      "url": "/noindex/css/fonts/Bold/OpenSans-Bold.ttf"
    },
    "rule": {
      "firedtimes": 4,
      "mail": false,
      "level": 10,
      "description": "IP address found in AlienVault reputation database.",
      "groups": [
        "attack"
      ],
      "id": "100100"
    },
    "location": "/var/log/httpd/access_log",
    "decoder": {
      "name": "web-accesslog"
    },
    "id": "1651055661.779757",
    "full_log": "192.168.33.1 - - [27/Apr/2022:12:34:20 +0200] \"GET /noindex/css/fonts/Bold/OpenSans-Bold.ttf HTTP/1.1\" 404 238 \"http://centos7/noindex/css/open-sans.css\" \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36\"",
    "timestamp": "2022-04-27T10:34:21.467+0000"
  },
  "fields": {
    "timestamp": [
      "2022-04-27T10:34:21.467Z"
    ]
  },
  "highlight": {
    "rule.id": [
      "@opensearch-dashboards-highlighted-field@100100@/opensearch-dashboards-highlighted-field@"
    ]
  },
  "sort": [
    1651055661467
  ]
}
```

</details>
<details><summary>Active response alert</summary>

```json
{
  "_index": "wazuh-alerts-4.x-2022.04.27",
  "_type": "_doc",
  "_id": "H1WWaoABKmeKtYx2JU0y",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "192.168.182.133",
      "name": "centos7",
      "id": "001"
    },
    "manager": {
      "name": "wazuh-manager"
    },
    "data": {
      "origin": {
        "module": "wazuh-execd",
        "name": "node01"
      },
      "version": "1",
      "parameters": {
        "alert": {
          "agent": {
            "ip": "192.168.182.133",
            "name": "centos7",
            "id": "001"
          },
          "manager": {
            "name": "wazuh-manager"
          },
          "data": {
            "protocol": "GET",
            "srcip": "192.168.33.1",
            "id": "404",
            "url": "/noindex/css/fonts/Light/OpenSans-Light.ttf"
          },
          "rule": {
            "firedtimes": "5",
            "mail": "false",
            "level": "10",
            "description": "IP address found in AlienVault reputation database.",
            "groups": [
              "attack"
            ],
            "id": "100100"
          },
          "location": "/var/log/httpd/access_log",
          "id": "1651055661.780233",
          "decoder": {
            "name": "web-accesslog"
          },
          "timestamp": "2022-04-27T10:34:21.467+0000",
          "full_log": "192.168.33.1 - - [27/Apr/2022:12:34:20 +0200] \"GET /noindex/css/fonts/Light/OpenSans-Light.ttf HTTP/1.1\" 404 240 \"http://centos7/noindex/css/open-sans.css\" \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36\""
        },
        "extra_args": [],
        "program": "active-response/bin/firewall-drop"
      },
      "command": "add"
    },
    "rule": {
      "firedtimes": 5,
      "mail": false,
      "level": 3,
      "pci_dss": [
        "11.4"
      ],
      "tsc": [
        "CC6.1",
        "CC6.8",
        "CC7.2",
        "CC7.3",
        "CC7.4"
      ],
      "description": "Host Blocked by firewall-drop Active Response",
      "groups": [
        "ossec",
        "active_response"
      ],
      "id": "651",
      "nist_800_53": [
        "SI.4"
      ],
      "gpg13": [
        "4.13"
      ],
      "gdpr": [
        "IV_35.7.d"
      ]
    },
    "location": "/var/ossec/logs/active-responses.log",
    "decoder": {
      "name": "ar_log_json"
    },
    "id": "1651055663.790719",
    "full_log": "2022/04/27 12:34:21 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-04-27T10:34:21.467+0000\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":5,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"001\",\"name\":\"centos7\",\"ip\":\"192.168.182.133\"},\"manager\":{\"name\":\"wazuh-manager\"},\"id\":\"1651055661.780233\",\"full_log\":\"192.168.33.1 - - [27/Apr/2022:12:34:20 +0200] \\\"GET /noindex/css/fonts/Light/OpenSans-Light.ttf HTTP/1.1\\\" 404 240 \\\"http://centos7/noindex/css/open-sans.css\\\" \\\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36\\\"\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.33.1\",\"id\":\"404\",\"url\":\"/noindex/css/fonts/Light/OpenSans-Light.ttf\"},\"location\":\"/var/log/httpd/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}",
    "timestamp": "2022-04-27T10:34:23.651+0000"
  },
  "fields": {
    "timestamp": [
      "2022-04-27T10:34:23.651Z"
    ]
  },
  "highlight": {
    "rule.id": [
      "@opensearch-dashboards-highlighted-field@651@/opensearch-dashboards-highlighted-field@"
    ]
  },
  "sort": [
    1651055663651
  ]
}
```

</details>

**Comment by vikman90** [comment]
## 🟢 Detecting and removing malware using VirusTotal integration

<details><summary>Agent setup</summary>

### Agent installation

```
wget https://packages-dev.wazuh.com/pre-release/apt/pool/main/w/wazuh-agent/wazuh-agent_4.3.0-1_amd64.deb
dpkg -i wazuh-agent_4.3.0-1_amd64.deb
apt-get install auditd
systemctl start wazuh-agent
```

### Agent configuration

```xml
<syscheck>
  <directories whodata="yes">/root</directories>
</syscheck>
```

### remove-threat.sh

```
nano /var/ossec/active-response/bin/remove-threat.sh
chmod 750 /var/ossec/active-response/bin/remove-threat.sh
chown root:wazuh /var/ossec/active-response/bin/remove-threat.sh
apt-get install jq
```

</details>
<details><summary>Manager setup</summary>

### Rules

```xml
<group name="syscheck,pci_dss_11.5,nist_800_53_SI.7,">
    <!-- Rules for Linux systems -->
    <rule id="100200" level="7">
        <if_sid>550</if_sid>
        <field name="file">/root</field>
        <description>File modified in /root directory.</description>
    </rule>
    <rule id="100201" level="7">
        <if_sid>554</if_sid>
        <field name="file">/root</field>
        <description>File added to /root directory.</description>
    </rule>
</group>

<group name="virustotal,">
  <rule id="100092" level="12">
      <if_sid>657</if_sid>
      <match>Successfully removed threat</match>
      <description>$(parameters.program) removed threat located at $(parameters.alert.data.virustotal.source.file)</description>
  </rule>

  <rule id="100093" level="12">
      <if_sid>657</if_sid>
      <match>Error removing threat</match>
      <description>Error removing threat located at $(parameters.alert.data.virustotal.source.file)</description>
  </rule>
</group>
```

### Decoder

```xml
<decoder name="ar_log_fields">
  <parent>ar_log</parent>
  <regex offset="after_parent">^(\S+) Removed threat located at (\S+)</regex>
  <order>script_name, path</order>
</decoder>
```

### Manager configuration

```xml
<ossec_config>
  <integration>
    <name>virustotal</name>
    <api_key>YOUR_VIRUS_TOTAL_API_KEY</api_key> <!-- Replace with your VirusTotal API key -->
    <rule_id>100200,100201</rule_id>
    <alert_format>json</alert_format>
  </integration>

  <command>
    <name>remove-threat</name>
    <executable>remove-threat.sh</executable>
    <timeout_allowed>no</timeout_allowed>
  </command>

  <active-response>
    <disabled>no</disabled>
    <command>remove-threat</command>
    <location>local</location>
    <rules_id>87105</rules_id>
  </active-response>
</ossec_config>
```

</details>

### Generating alerts

```sh
cd /root
curl -LO http://www.eicar.org/download/eicar.com && ls -lah eicar.com
```

### Results

#### Discovery query screenshot

![discover-threat](https://user-images.githubusercontent.com/10536251/165512216-ae1fe78e-ef9f-4ec4-a099-de67cd5cf794.jpg)

<details><summary>Threat deletion alert</summary>

```json
{
  "_index": "wazuh-alerts-4.x-2022.04.27",
  "_type": "_doc",
  "_id": "XFW_aoABKmeKtYx2ak4z",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "192.168.182.134",
      "name": "focal",
      "id": "002"
    },
    "manager": {
      "name": "wazuh-manager"
    },
    "data": {
      "origin": {
        "module": "wazuh-execd",
        "name": "node01"
      },
      "version": "1",
      "parameters": {
        "alert": {
          "agent": {
            "ip": "192.168.182.134",
            "name": "focal",
            "id": "002"
          },
          "manager": {
            "name": "wazuh-manager"
          },
          "data": {
            "integration": "virustotal",
            "virustotal": {
              "sha1": "3395856ce81f2b7382dee72602f798b642f14140",
              "malicious": "1",
              "total": "65",
              "found": "1",
              "positives": "60",
              "source": {
                "sha1": "3395856ce81f2b7382dee72602f798b642f14140",
                "file": "/root/eicar.com",
                "alert_id": "1651058364.1422431",
                "md5": "44d88612fea8a8f36de82e1278abb02f"
              },
              "permalink": "https://www.virustotal.com/gui/file/275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f/detection/f-275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f-1651057738",
              "scan_date": "2022-04-27 11:08:58"
            }
          },
          "rule": {
            "firedtimes": "1",
            "mail": "true",
            "level": "12",
            "pci_dss": [
              "10.6.1",
              "11.4"
            ],
            "description": "VirusTotal: Alert - /root/eicar.com - 60 engines detected this file",
            "groups": [
              "virustotal"
            ],
            "mitre": {
              "technique": [
                "Exploitation for Client Execution"
              ],
              "id": [
                "T1203"
              ],
              "tactic": [
                "Execution"
              ]
            },
            "id": "87105",
            "gdpr": [
              "IV_35.7.d"
            ]
          },
          "location": "virustotal",
          "id": "1651058365.1422431",
          "decoder": {
            "name": "json"
          },
          "timestamp": "2022-04-27T11:19:25.771+0000"
        },
        "extra_args": [],
        "program": "active-response/bin/remove-threat.sh"
      },
      "command": "add"
    },
    "rule": {
      "firedtimes": 1,
      "mail": true,
      "level": 12,
      "description": "active-response/bin/remove-threat.sh removed threat located at /root/eicar.com",
      "groups": [
        "virustotal"
      ],
      "id": "100092"
    },
    "location": "/var/ossec/logs/active-responses.log",
    "decoder": {
      "name": "ar_log_json"
    },
    "id": "1651058366.1425746",
    "full_log": "2022/04/27 13:19:25 active-response/bin/remove-threat.sh: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-04-27T11:19:25.771+0000\",\"rule\":{\"level\":12,\"description\":\"VirusTotal: Alert - /root/eicar.com - 60 engines detected this file\",\"id\":\"87105\",\"mitre\":{\"id\":[\"T1203\"],\"tactic\":[\"Execution\"],\"technique\":[\"Exploitation for Client Execution\"]},\"firedtimes\":1,\"mail\":true,\"groups\":[\"virustotal\"],\"pci_dss\":[\"10.6.1\",\"11.4\"],\"gdpr\":[\"IV_35.7.d\"]},\"agent\":{\"id\":\"002\",\"name\":\"focal\",\"ip\":\"192.168.182.134\"},\"manager\":{\"name\":\"wazuh-manager\"},\"id\":\"1651058365.1422431\",\"decoder\":{\"name\":\"json\"},\"data\":{\"virustotal\":{\"found\":\"1\",\"malicious\":\"1\",\"source\":{\"alert_id\":\"1651058364.1422431\",\"file\":\"/root/eicar.com\",\"md5\":\"44d88612fea8a8f36de82e1278abb02f\",\"sha1\":\"3395856ce81f2b7382dee72602f798b642f14140\"},\"sha1\":\"3395856ce81f2b7382dee72602f798b642f14140\",\"scan_date\":\"2022-04-27 11:08:58\",\"positives\":\"60\",\"total\":\"65\",\"permalink\":\"https://www.virustotal.com/gui/file/275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f/detection/f-275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f-1651057738\"},\"integration\":\"virustotal\"},\"location\":\"virustotal\"},\"program\":\"active-response/bin/remove-threat.sh\"}} Successfully removed threat",
    "timestamp": "2022-04-27T11:19:26.861+0000"
  },
  "fields": {
    "timestamp": [
      "2022-04-27T11:19:26.861Z"
    ]
  },
  "highlight": {
    "rule.description": [
      "@opensearch-dashboards-highlighted-field@active-response/bin/remove-threat.sh removed threat located at /root/eicar.com@/opensearch-dashboards-highlighted-field@"
    ],
    "full_log": [
      "2022/04/27 13:19:25 active-response/bin/remove-threat.sh: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-04-27T11:19:25.771+0000\",\"rule\":{\"level\":12,\"description\":\"VirusTotal: Alert - /root/@opensearch-dashboards-highlighted-field@eicar.com@/opensearch-dashboards-highlighted-field@ - 60 engines detected this file\",\"id\":\"87105\",\"mitre\":{\"id\":[\"T1203\"],\"tactic\":[\"Execution\"],\"technique\":[\"Exploitation for Client Execution\"]},\"firedtimes\":1,\"mail\":true,\"groups\":[\"virustotal\"],\"pci_dss\":[\"10.6.1\",\"11.4\"],\"gdpr\":[\"IV_35.7.d\"]},\"agent\":{\"id\":\"002\",\"name\":\"focal\",\"ip\":\"192.168.182.134\"},\"manager\":{\"name\":\"wazuh-manager\"},\"id\":\"1651058365.1422431\",\"decoder\":{\"name\":\"json\"},\"data\":{\"virustotal\":{\"found\":\"1\",\"malicious\":\"1\",\"source\":{\"alert_id\":\"1651058364.1422431\",\"file\":\"/root/@opensearch-dashboards-highlighted-field@eicar.com@/opensearch-dashboards-highlighted-field@\",\"md5\":\"44d88612fea8a8f36de82e1278abb02f\",\"sha1\":\"3395856ce81f2b7382dee72602f798b642f14140\"},\"sha1\":\"3395856ce81f2b7382dee72602f798b642f14140\",\"scan_date\":\"2022-04-27 11:08:58\",\"positives\":\"60\",\"total\":\"65\",\"permalink\":\"https://www.virustotal.com/gui/file/275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f/detection/f-275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f-1651057738\"},\"integration\":\"virustotal\"},\"location\":\"virustotal\"},\"program\":\"active-response/bin/remove-threat.sh\"}} Successfully removed threat"
    ]
  },
  "sort": [
    1651058366861
  ]
}
```

</details>

**Comment by vikman90** [comment]
🟡 Detecting malware using Yara integration

<details><summary>Manager setup</summary>

### Rules

```xml
<group name="syscheck,">
  <rule id="100300" level="7">
    <if_sid>550</if_sid>
    <field name="file">/tmp/yara/malware/</field>
    <description>File modified in /tmp/yara/malware/ directory.</description>
  </rule>
  <rule id="100301" level="7">
    <if_sid>554</if_sid>
    <field name="file">/tmp/yara/malware/</field>
    <description>File added to /tmp/yara/malware/ directory.</description>
  </rule>
</group>

<group name="yara,">
  <rule id="108000" level="0">
    <decoded_as>yara_decoder</decoded_as>
    <description>Yara grouping rule</description>
  </rule>
  <rule id="108001" level="12">
    <if_sid>108000</if_sid>
    <match>wazuh-yara: INFO - Scan result: </match>
    <description>File "$(yara_scanned_file)" is a positive match. Yara rule: $(yara_rule)</description>
  </rule>
</group>
```

### Decoder

```xml
<decoder name="yara_decoder">
  <prematch>wazuh-yara:</prematch>
</decoder>

<decoder name="yara_decoder1">
  <parent>yara_decoder</parent>
  <regex>wazuh-yara: (\S+) - Scan result: (\S+) (\S+)</regex>
  <order>log_type, yara_rule, yara_scanned_file</order>
</decoder>
```

### Manager configuration

```xml
<ossec_config>
  <command>
    <name>yara</name>
    <executable>yara.sh</executable>
    <extra_args>-yara_path /usr/local/bin -yara_rules /tmp/yara/rules/yara_rules.yar</extra_args>
    <timeout_allowed>no</timeout_allowed>
  </command>
  <active-response>
    <command>yara</command>
    <location>local</location>
    <rules_id>100300,100301</rules_id>
  </active-response>
</ossec_config>
```

</details>
<details><summary>Agent setup</summary>

### Installation

```shell
wget https://packages-dev.wazuh.com/pre-release/apt/pool/main/w/wazuh-agent/wazuh-agent_4.3.0-1_amd64.deb
dpkg -i wazuh-agent_4.3.0-1_amd64.deb
systemctl start wazuh-agent

apt update
apt install -y make gcc autoconf libtool libssl-dev pkg-config
curl -LO https://github.com/VirusTotal/yara/archive/v4.0.2.tar.gz
tar -xvzf v4.0.2.tar.gz -C /usr/local/bin/ && rm -f v4.0.2.tar.gz
cd /usr/local/bin/yara-4.0.2
./bootstrap.sh && ./configure && make && sudo make install && make check

mkdir -p /tmp/yara/rules
curl 'https://valhalla.nextron-systems.com/api/v1/get' \
-H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' \
-H 'Accept-Language: en-US,en;q=0.5' \
--compressed \
-H 'Referer: https://valhalla.nextron-systems.com/' \
-H 'Content-Type: application/x-www-form-urlencoded' \
-H 'DNT: 1' -H 'Connection: keep-alive' -H 'Upgrade-Insecure-Requests: 1' \
--data 'demo=demo&apikey=1111111111111111111111111111111111111111111111111111111111111111&format=text' \
-o /tmp/yara/rules/yara_rules.yar

apt install -y jq
```

### yara.sh

```shell
nano /var/ossec/active-response/bin/yara.sh
chown root:wazuh /var/ossec/active-response/bin/yara.sh
chmod 750 /var/ossec/active-response/bin/yara.sh
```

### Agent configuration

```xml
<syscheck>
  <directories whodata="yes" realtime="yes">/tmp/yara/malware</directories>
</syscheck>
```

</details>

### Producing alerts

- Download /tmp/yara/malware/malware_downloader.sh

```shell
bash /tmp/yara/malware/malware_downloader.sh
```

### Results

#### Discover screenshot

![discover-yara](https://user-images.githubusercontent.com/10536251/165516050-b7601160-a0de-4c47-80b1-e8a318279e1f.jpg)

<details><summary>YARA alerts</summary>

```
** Alert 1651061533.1531783: mail  - yara,
2022 Apr 27 12:12:13 (focal) any->/var/ossec/logs/active-responses.log
Rule: 108001 (level 12) -> 'File "/tmp/yara/malware/xbash" is a positive match. Yara rule: MAL_Xbash_PY_Sep18_RID2D38'
wazuh-yara: INFO - Scan result: MAL_Xbash_PY_Sep18_RID2D38 /tmp/yara/malware/xbash
log_type: INFO
yara_rule: MAL_Xbash_PY_Sep18_RID2D38
yara_scanned_file: /tmp/yara/malware/xbash

** Alert 1651061533.1532196: mail  - yara,
2022 Apr 27 12:12:13 (focal) any->/var/ossec/logs/active-responses.log
Rule: 108001 (level 12) -> 'File "/tmp/yara/malware/vpn_filter" is a positive match. Yara rule: MAL_ELF_VPNFilter_3_RID2D6C'
wazuh-yara: INFO - Scan result: MAL_ELF_VPNFilter_3_RID2D6C /tmp/yara/malware/vpn_filter
log_type: INFO
yara_rule: MAL_ELF_VPNFilter_3_RID2D6C
yara_scanned_file: /tmp/yara/malware/vpn_filter

** Alert 1651061535.1532627: mail  - yara,
2022 Apr 27 12:12:15 (focal) any->/var/ossec/logs/active-responses.log
Rule: 108001 (level 12) -> 'File "/tmp/yara/malware/webshell" is a positive match. Yara rule: Webshell_Worse_Linux_Shell_1_RID320C'
wazuh-yara: INFO - Scan result: Webshell_Worse_Linux_Shell_1_RID320C /tmp/yara/malware/webshell
log_type: INFO
yara_rule: Webshell_Worse_Linux_Shell_1_RID320C
yara_scanned_file: /tmp/yara/malware/webshell

** Alert 1651061535.1533079: mail  - yara,
2022 Apr 27 12:12:15 (focal) any->/var/ossec/logs/active-responses.log
Rule: 108001 (level 12) -> 'File "/tmp/yara/malware/webshell" is a positive match. Yara rule: Webshell_Worse_Linux_Shell_php_RID3323'
wazuh-yara: INFO - Scan result: Webshell_Worse_Linux_Shell_php_RID3323 /tmp/yara/malware/webshell
log_type: INFO
yara_rule: Webshell_Worse_Linux_Shell_php_RID3323
yara_scanned_file: /tmp/yara/malware/webshell
```

</details>

### Issues

- Enabling real-time and who-data together is superfluous.

**Comment by vikman90** [comment]
## 🟢 How to integrate YARA with Wazuh

As stated at #12870, the blogpost content is **deprecated**, but it points to the use case [How to integrate Wazuh with YARA](https://documentation-dev.wazuh.com/current/user-manual/capabilities/active-response/ar-use-cases/wazuh-with-yara.html#how-to-integrate-wazuh-with-yara), which we will follow here.

<details><summary>Manager setup</summary>

### Manager configuration

```xml
<ossec_config>
  <command>
    <name>yara</name>
    <executable>yara.sh</executable>
    <extra_args>-yara_path /path/to/yara -yara_rules /tmp/yara/rules/hiddenwasp.yar</extra_args>
    <timeout_allowed>no</timeout_allowed>
  </command>
  <active-response>
    <command>yara</command>
    <location>local</location>
    <rules_id>550,554</rules_id>
  </active-response>
</ossec_config>
```

### Decoder

```xml
<!--
 - YARA decoders
 - Created by Wazuh, Inc.
 - Copyright (C) 2015-2022, Wazuh Inc.
 - This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2.
-->

<decoder name="yara">
  <prematch>wazuh-yara: </prematch>
</decoder>

<decoder name="yara">
  <parent>yara</parent>
  <regex>wazuh-yara: (\S+) - Scan result: (\S+) (\S+)</regex>
  <order>log_type, yara_rule, yara_scanned_file</order>
</decoder>

<decoder name="yara">
  <parent>yara</parent>
  <regex>wazuh-yara: (\S+) - Yara active response error. (\.+)</regex>
  <order>log_type, error_message</order>
</decoder>
```

### Rules

```xml
<!--
 - YARA rules
 - Created by Wazuh, Inc.
 - Copyright (C) 2015-2022, Wazuh Inc.
 - This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2.
-->

 <group name="yara,">
    <rule id="100100" level="0">
        <decoded_as>yara</decoded_as>
        <description>YARA rules grouped.</description>
    </rule>

    <rule id="100101" level="5">
        <if_sid>100100</if_sid>
        <field name="error_message">\.+</field>
        <description>YARA error detected.</description>
    </rule>

    <rule id="100102" level="10">
        <if_sid>100100</if_sid>
        <field name="yara_rule">\.+</field>
        <description>YARA $(yara_rule) detected.</description>
    </rule>
</group>
```

</details>
<details><summary>Agent setup</summary>

### Agent configuration

```xml
<syscheck>
  <directories whodata="yes" realtime="yes">/tmp/yara/malware</directories>
</syscheck>
```


### yara.sh

```shell
nano /var/ossec/active-response/bin/yara.sh
chown root:wazuh /var/ossec/active-response/bin/yara.sh
chmod 750 /var/ossec/active-response/bin/yara.sh
apt-get install -y jq
mkdir -p /tmp/yara/rules
nano /tmp/yara/rules/hiddenwasp.yar
```

</details>

### Producing events

```shell
echo "curl 103.206.123.13" > /tmp/yara/malware/hiddenwasp.sh
```

### Result

#### Alert screenshot

![screenshot yara](https://user-images.githubusercontent.com/10536251/165754890-3f4cb237-3502-47b5-8d09-35a76c6e8892.jpg)

<details><summary>JSON alert</summary>

```json
{
  "_index": "wazuh-alerts-4.x-2022.04.28",
  "_type": "_doc",
  "_id": "LtItcIABzSqndtCMRs_n",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "log"
    },
    "agent": {
      "ip": "192.168.182.134",
      "name": "focal",
      "id": "002"
    },
    "manager": {
      "name": "wazuh-manager"
    },
    "data": {
      "log_type": "INFO",
      "yara_rule": "HiddenWasp_Deployment",
      "yara_scanned_file": "/tmp/yara/malware/hiddenwasp.sh"
    },
    "rule": {
      "firedtimes": 1,
      "mail": false,
      "level": 10,
      "description": "YARA HiddenWasp_Deployment detected.",
      "groups": [
        "yara"
      ],
      "id": "100102"
    },
    "location": "/var/ossec/logs/active-responses.log",
    "decoder": {
      "name": "yara"
    },
    "id": "1651149450.301377",
    "full_log": "wazuh-yara: INFO - Scan result: HiddenWasp_Deployment /tmp/yara/malware/hiddenwasp.sh",
    "timestamp": "2022-04-28T12:37:30.889+0000"
  },
  "fields": {
    "timestamp": [
      "2022-04-28T12:37:30.889Z"
    ]
  },
  "highlight": {
    "rule.id": [
      "@opensearch-dashboards-highlighted-field@100102@/opensearch-dashboards-highlighted-field@"
    ]
  },
  "sort": [
    1651149450889
  ]
}
```

</details>
